src/state/opamState.ml

```diff
@@ -1292,6 +1297,24 @@ let dump_state t oc =
         (string_of_conjunction (fun a -> a) m);
     with Not_found -> () );
 
+    (match repository_and_prefix_of_package t package with
```

Comment:
Why do you read the opam files again rather than accessing `t.opams` ? You should use function `OpamState.opam`, unless you specifically want to skip pinned package's overlays, and `OpamPackage.Map.find pkg t.opams` otherwise.